test/components/rankers/test_sentence_transformers_diversity.py

```diff
@@ -0,0 +1,394 @@
+import pytest
+
+from haystack import ComponentError, Document
+from haystack.components.rankers import SentenceTransformersDiversityRanker
+from haystack.utils import ComponentDevice
+from haystack.utils.auth import Secret
+
+
+class TestSentenceTransformersDiversityRanker:
+    def test_init(self):
+        component = SentenceTransformersDiversityRanker()
+        assert component.model_name_or_path == "sentence-transformers/all-MiniLM-L6-v2"
+        assert component.top_k == 10
+        assert component.device == ComponentDevice.resolve_device(None)
+        assert component.similarity == "cosine"
+        assert component.token == Secret.from_env_var("HF_API_TOKEN", strict=False)
+        assert component.query_prefix == ""
+        assert component.document_prefix == ""
+        assert component.query_suffix == ""
+        assert component.document_suffix == ""
+        assert component.meta_fields_to_embed == []
+        assert component.embedding_separator == "\n"
+
+    def test_init_with_custom_init_parameters(self):
+        component = SentenceTransformersDiversityRanker(
+            model="sentence-transformers/msmarco-distilbert-base-v4",
+            top_k=5,
+            device=ComponentDevice.from_str("cuda:0"),
+            token=Secret.from_token("fake-api-token"),
+            similarity="dot_product",
+            query_prefix="query:",
+            document_prefix="document:",
+            query_suffix="query suffix",
+            document_suffix="document suffix",
+            meta_fields_to_embed=["meta_field"],
+            embedding_separator="--",
+        )
+        assert component.model_name_or_path == "sentence-transformers/msmarco-distilbert-base-v4"
+        assert component.top_k == 5
+        assert component.device == ComponentDevice.from_str("cuda:0")
+        assert component.similarity == "dot_product"
+        assert component.token == Secret.from_token("fake-api-token")
+        assert component.query_prefix == "query:"
+        assert component.document_prefix == "document:"
+        assert component.query_suffix == "query suffix"
+        assert component.document_suffix == "document suffix"
+        assert component.meta_fields_to_embed == ["meta_field"]
+        assert component.embedding_separator == "--"
+
+    def test_to_and_from_dict(self):
```

Comment:
To be consistent with other tests for other components could we split this test into separate `test_to_dict` and `test_from_dict`?